client/block.c

```diff
@@ -1902,3 +1902,9 @@ void xdag_list_orphan_blocks(int count, FILE *out)
 
 	pthread_mutex_unlock(&block_mutex);
 }
+
+// completes work with the blocks
+void xdag_block_finish(void)
+{
+	pthread_mutex_lock(&block_mutex);
```

Comment:
I donot think it will solve this problem that already tried such fix one month ago. Better test it before submit.

Reply:
don't know, on my system works.. i don't have this issue anymore
which test did you do ?

Reply:
Please see issue #191 #192 , those are issues I met with Mac, and I did some fixes for those issues, but not working at all. xdag_block_finish was included in my fixes, so I didnot submit those code changes.

Reply:
I never had #192 issue.
My issue is that:
*Without -z RAM* it crash if i terminate while loading from disk (almost always) and very often while sync or normal run. Basically i always had to manually kill xdag to terminate it.
And i don’t have this issues anymore now.
But maybe the issues that you reported on mac are different. For example i never saw #192.

Reply:
oh yes, i found another issue